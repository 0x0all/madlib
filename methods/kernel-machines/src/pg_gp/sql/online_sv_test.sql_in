-- Example usage for regression:
-- \i online_sv.sql_in

select MADLIB_SCHEMA.svm_generate_reg_data('MADLIB_SCHEMA.svm_train_data', 1000, 5);
select MADLIB_SCHEMA.svm_regression('MADLIB_SCHEMA.svm_train_data', 'myexp', false);
select MADLIB_SCHEMA.svm_predict('myexp', '{1,2,4,20,10}') > 0;
select MADLIB_SCHEMA.svm_predict('myexp', '{1,2,4,20,-10}') < 0;

   
-- To learn multiple support vector regression models
select MADLIB_SCHEMA.svm_regression('MADLIB_SCHEMA.svm_train_data', 'myexp', true);
select pred.prediction > 0 from MADLIB_SCHEMA.svm_predict_combo('myexp', '{1,2,4,20,10}') as pred;

-- Example usage for classification:
select MADLIB_SCHEMA.svm_generate_cls_data('MADLIB_SCHEMA.svm_train_data', 1000, 2);
select MADLIB_SCHEMA.svm_classification('MADLIB_SCHEMA.svm_train_data', 'myexpc', false);
select MADLIB_SCHEMA.svm_predict('myexpc', '{10,-20}') > 0;
select MADLIB_SCHEMA.svm_predict('myexpc', '{-10,20}') < 0;


-- To learn multiple support vector models, replace the above by 
select MADLIB_SCHEMA.svm_classification('MADLIB_SCHEMA.svm_train_data', 'myexpc', true);
select pred.prediction > 0 from MADLIB_SCHEMA.svm_predict_combo('myexpc', '{10,-20}') as pred;
select pred.prediction < 0 from MADLIB_SCHEMA.svm_predict_combo('myexpc', '{-10,20}') as pred;

-- Example usage for novelty detection:
select MADLIB_SCHEMA.svm_generate_nd_data('MADLIB_SCHEMA.svm_train_data', 500, 2);
select MADLIB_SCHEMA.svm_novelty_detection('MADLIB_SCHEMA.svm_train_data', 'myexpnd', false);
select MADLIB_SCHEMA.svm_predict('myexpnd', '{10,-5}') > 0;  
select MADLIB_SCHEMA.svm_predict('myexpnd', '{-1,-1}') < 0;  

