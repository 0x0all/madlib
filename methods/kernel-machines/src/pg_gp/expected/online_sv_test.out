-- Example usage for regression:
-- \i online_sv.sql_in
select madlib.svm_generate_reg_data('madlib.svm_train_data', 1000, 5);
 svm_generate_reg_data 
-----------------------
 
(1 row)

select madlib.svm_regression('madlib.svm_train_data', 'myexp', false);
                                svm_regression                                
------------------------------------------------------------------------------
 Finished support vector regression learning on madlib.svm_train_data table. 
            
(1 row)

select madlib.svm_predict('myexp', '{1,2,4,20,10}') > 0;
 ?column? 
----------
 t
(1 row)

select madlib.svm_predict('myexp', '{1,2,4,20,-10}') < 0;
 ?column? 
----------
 t
(1 row)

   
-- To learn multiple support vector regression models
select madlib.svm_regression('madlib.svm_train_data', 'myexp', true);
                                svm_regression                                
------------------------------------------------------------------------------
 Finished support vector regression learning on madlib.svm_train_data table. 
            
(1 row)

select pred.prediction > 0 from madlib.svm_predict_combo('myexp', '{1,2,4,20,10}') as pred;
 ?column? 
----------
 t
 t
 t
(3 rows)

-- Example usage for classification:
select madlib.svm_generate_cls_data('madlib.svm_train_data', 1000, 2);
 svm_generate_cls_data 
-----------------------
 
(1 row)

select madlib.svm_classification('madlib.svm_train_data', 'myexpc', false);
                                svm_classification                                
----------------------------------------------------------------------------------
 Finished support vector classification learning on madlib.svm_train_data table. 
            
(1 row)

select madlib.svm_predict('myexpc', '{10,-20}') > 0;
 ?column? 
----------
 t
(1 row)

select madlib.svm_predict('myexpc', '{-10,20}') < 0;
 ?column? 
----------
 t
(1 row)

-- To learn multiple support vector models, replace the above by 
select madlib.svm_classification('madlib.svm_train_data', 'myexpc', true);
                                svm_classification                                
----------------------------------------------------------------------------------
 Finished support vector classification learning on madlib.svm_train_data table. 
            
(1 row)

select pred.prediction > 0 from madlib.svm_predict_combo('myexpc', '{10,-20}') as pred;
 ?column? 
----------
 t
 t
 t
(3 rows)

select pred.prediction < 0 from madlib.svm_predict_combo('myexpc', '{-10,20}') as pred;
 ?column? 
----------
 t
 t
 t
(3 rows)

-- Example usage for novelty detection:
select madlib.svm_generate_nd_data('madlib.svm_train_data', 500, 2);
 svm_generate_nd_data 
----------------------
 
(1 row)

select madlib.svm_novelty_detection('madlib.svm_train_data', 'myexpnd', false);
                                svm_novelty_detection                                
-------------------------------------------------------------------------------------
 Finished support vector novelty detection learning on madlib.svm_train_data table. 
            
(1 row)

select madlib.svm_predict('myexpnd', '{10,-5}') > 0;  
 ?column? 
----------
 t
(1 row)

select madlib.svm_predict('myexpnd', '{-1,-1}') < 0;  
 ?column? 
----------
 t
(1 row)

